sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -388,6 +388,23 @@ def _create_pipeline_workflow(self, args, pipeline, op_transformers=None, pipeli
       if pipeline_conf.timeout:
         pipelinerun['spec']['timeout'] = '%ds' % pipeline_conf.timeout
 
+      # generate the Tekton service account template
+      service_template = {}
+      if len(pipeline_conf.image_pull_secrets) > 0:
+        service_template = {
+          'apiVersion': 'v1',
+          'kind': 'ServiceAccount',
+          'metadata': {'name': 'secrets'}
+        }
+      image_pull_secrets = []
+      for image_pull_secret in pipeline_conf.image_pull_secrets:
+        image_pull_secrets.append(convert_k8s_obj_to_json(image_pull_secret))
```

Comment:
It looks like `image_pull_secrets` is not being used for service_template, maybe you can remove this line.

Reply:
ok